PKHeX/Legality/Checks.cs

```diff
@@ -791,6 +791,7 @@ private CheckResult verifyEncounterG3Transfer()
             CheckResult InvalidTransferResult = null;
             CheckResult EggResult = null;
             CheckResult G3Result = null;
+            object G3Encounter = null;
```

Comment:
G3Encounter cant be of type CheckResult, is not to store the result, it should store the encounter to have EncounterMatch fill after the function,

EncounterMatch is declared as object because it can contain a variable of type EncounterSlot, EncounterStatic, EncounterTrade or int (for eggs)